File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
    if (input.toUpperCase() == input && input.toLowerCase() != input){
        return "Whoa, chill out!";
    }
    if (input.endsWith("?")){
        return "Sure.";
    }
    if (input.trim() == ""){
        return "Fine. Be that way!";
    }
    return "Whatever."
};

module.exports = Bob;

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with?(input, "?") -> "Sure." 
      String.upcase(input) == input && String.downcase(input) != input -> "Whoa, chill out!"
      String.strip(input) == "" -> "Fine. Be that way!"
      true  -> "Whatever."
    end
  end
end
